config/hooks/pre-commit

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+set -e
+
+REPO_ROOT_DIR="$(git rev-parse --show-toplevel)"
+
+files=$((git diff --cached --name-only --diff-filter=ACMR | grep -Ei "\.java$") || true)
```

Comment:
Shells disambiguate `$((` differently or not at all. Add space, replacing `$((` with `$( (`.

Caught by `shellcheck` linter.